documentation/develop.md

```diff
@@ -0,0 +1,86 @@
+# Development
+
+Please as a first start, tell the community about your intent of developing a feature or fix a bug. Search for the associated issue if it exists or open one with steps to reproduce (for bugs) or a [user story](https://en.wikipedia.org/wiki/User_story#Principle) (for features).
+
+## Setup
+
+### Pre-requisites
+
+To setup your environment, you would need to install the following dependencies:
+ - git
+ - [nvm](https://github.com/nvm-sh/nvm/blob/master/README.md)
+ - Firefox and/or Chromium to run the end-to-end (nbrowsers) tests
+ - Python (preferably Python 3.9) and virtualenv
+    - :warning: As of 2023-06-06, Python 3.11 is not supported due to the version of the [wrapt dependency](https://github.com/GrahamDumpleton/wrapt/issues/196)
+
+### Clone the repository
+
+```bash
+$ git clone https://github.com/gristlabs/grist-core
+```
+
+Then go to the grist-core root directory before install nodejs through nvm:
+
+```bash
+$ cd grist-core/
+```
+
+### Setup nodejs
+
+You need to install the supported nodejs version as well as yarn. To do so, at the grist-core root directory, run the following command to install nodejs via nvm:
+
+```bash
+$ nvm install
+```
+
+Now check that node is installed in the version specified in the `.nvmrc` file:
+
+```bash
+$ node --version
+```
+
+Then install yarn (the `-g` flag here means that yarn will be available globally):
+```bash
+$ npm install -g yarn
+```
+
+Now each time you want to load nodejs and yarn in your environment, just run the following command at grist-core root directory:
+
+```bash
+$ nvm use
+```
+
+### Install the python dependencies
+
+Ensure to have Python and virtualenv installed. On debian-based Linux distributions, you can simply run the following command as root:
+
+```bash
+# apt install python3.9 python3.9-venv
+```
+
+## Start the server in development mode
+
+Just run the following command:
+```bash
+$ yarn start
+```
+
+Each time you change something, just reload the webpage in your browser.
+
+Happy coding!
+
+### Pick an issue
+
+Lost on what you can do to help? If you are new to Grist, you may just pick one of the issues labelled `good first issue`:
+
+https://github.com/gristlabs/grist-core/labels/good%20first%20issue
+
+## Develop widgets
+
+TODO
```

Comment:
If this is about custom widgets, it could point to https://github.com/gristlabs/grist-widget#readme or https://support.getgrist.com/widget-custom/ (though I do see looking at this now that neither is ideal).